pkg/knuu/knuu.go

```diff
@@ -130,18 +141,23 @@ func handleTimeout() error {
 	if err := instance.Commit(); err != nil {
 		return fmt.Errorf("cannot commit instance: %s", err)
 	}
-	timeoutSeconds := int64(timeout.Seconds())
-
-	// command to wait for timeout and delete all resources with the identifier
-	var command = []string{"sh", "-c"}
-	// Command runs in-cluster to delete resources post-test. Chosen for simplicity over a separate Go app.
-	wait := fmt.Sprintf("sleep %d", timeoutSeconds)
-	deleteAllButTimeOutType := fmt.Sprintf("kubectl get all,pvc,netpol,roles,serviceaccounts,rolebindings,configmaps -l knuu.sh/test-run-id=%s -n %s -o json | jq -r '.items[] | select(.metadata.labels.\"knuu.sh/type\" != \"%s\") | \"\\(.kind)/\\(.metadata.name)\"' | xargs -r kubectl delete -n %s", identifier, k8s.Namespace(), TimeoutHandlerInstance.String(), k8s.Namespace())
-	deleteAll := fmt.Sprintf("kubectl delete all,pvc,netpol,roles,serviceaccounts,rolebindings,configmaps -l knuu.sh/test-run-id=%s -n %s", identifier, k8s.Namespace())
-	cmd := fmt.Sprintf("%s && %s && %s", wait, deleteAllButTimeOutType, deleteAll)
-	command = append(command, cmd)
-
-	if err := instance.SetCommand(command...); err != nil {
+
+	// Check if KNUU_DEDICATED_NAMESPACE is true
+	useDedicatedNamespace, _ := strconv.ParseBool(os.Getenv("KNUU_DEDICATED_NAMESPACE"))
+
+	var commands []string
+	commands = append(commands, fmt.Sprintf("sleep %d", int64(timeout.Seconds())))
+	commands = append(commands, fmt.Sprintf("kubectl get all,pvc,netpol,roles,serviceaccounts,rolebindings,configmaps -l knuu.sh/test-run-id=%s -n %s -o json | jq -r '.items[] | select(.metadata.labels.\"knuu.sh/type\" != \"%s\") | \"\\(.kind)/\\(.metadata.name)\"' | xargs -r kubectl delete -n %s", identifier, k8s.Namespace(), TimeoutHandlerInstance.String(), k8s.Namespace()))
+
+	if useDedicatedNamespace {
+		commands = append(commands, fmt.Sprintf("kubectl delete namespace %s", k8s.Namespace()))
+	}
+
+	commands = append(commands, fmt.Sprintf("kubectl delete all,pvc,netpol,roles,serviceaccounts,rolebindings,configmaps -l knuu.sh/test-run-id=%s -n %s", identifier, k8s.Namespace()))
```

Comment:
doc: Please add comments to each line to explain what the commands are doing so that it's not necessary to understand the commands themselves.